src/main/java/core/basesyntax/dao/ReportExporterDaoImpl.java

```
@@ -0,0 +1,32 @@
+package core.basesyntax.dao;
+import core.basesyntax.exeptions.FileWriteException;
+import core.basesyntax.exeptions.IncorrectFileNameException;
+import core.basesyntax.exeptions.IncorrectFormatOfReportException;
+import core.basesyntax.exeptions.ReportIsNullException;
+import java.io.FileWriter;
+import java.io.IOException;
+public class ReportExporterDaoImpl implements ReportExporterDao {
+    private final String fileName;
+
+    public ReportExporterDaoImpl(String fileName) {
+        if (fileName == null || fileName.isBlank()) {
+            throw new IncorrectFileNameException("File name is incorrect " + fileName);
+        }
+        this.fileName = fileName;
+    }
+
+    @Override
+    public void writeTheReport(String report) {
+        if (report == null) {
+            throw new ReportIsNullException("Report can't be null.");
+        }
+        if (report.isBlank()) {
+            throw new IncorrectFormatOfReportException("Report is incorrect " + fileName);
+        }
+        try (FileWriter fileWriter = new FileWriter(fileName)) {
+            fileWriter.write(report);
+        } catch (IOException e) {
+            throw new FileWriteException("Error writing report to " + fileName, e);
```

The code wraps the IOException and rethrows a FileWriteException here. Ensure FileWriteException(String, Throwable) preserves the original cause (calls super(message, cause)). Currently the FileWriteException implementation discards the passed cause which loses the original IOException; this violates the checklist guidance: "Don't ignore exceptions in try-catch blocks... Let's rethrow a RuntimeException with an **informative** message and exception object." Update the FileWriteException constructors accordingly so the original cause is preserved when rethrowing. See FileWriteException implementation for the incorrect constructor behavior.